chore: create datadog-flagging package (FFL-23)

leoromanovsky opened the pull request:
## Motivation

<!-- Why are you making this change, what problem does it solve? Include links to relevant tickets. -->

## Changes

<!-- What does this change exactly? Who will be affected? Include relevant screenshots, videos, links. Please highlight all the changes that you are not sure about (ex: AI agent generated) -->



## Test instructions

<!-- How can the reviewer test this change? Include relevant steps to reproduce the issue, if any. -->

Run `yarn build && yarn dev` to start the development server locally.

Navigate to `http://localhost:8080/` and observe the "cdn" served page:

<img width="1052" alt="Screenshot 2025-05-21 at 9 48 11 PM" src="https://github.com/user-attachments/assets/564529b6-1784-4122-9057-8abfe081f45a" />
<img width="1344" alt="Screenshot 2025-05-21 at 10 07 20 PM" src="https://github.com/user-attachments/assets/c3da7173-a727-44fd-a976-a0538e451ead" />

Next navigate to `http://localhost:8080/react-app/` to observe the npm build page:

<img width="1052" alt="Screenshot 2025-05-21 at 10 07 00 PM" src="https://github.com/user-attachments/assets/cbf12022-246a-4bea-947a-455b89bcef6d" />

## Checklist

<!-- By submitting this test, you confirm the following: -->

- [ ] Tested locally
- [ ] Tested on staging
- [ ] Added unit tests for this change.
- [ ] Added e2e/integration tests for this change.

<!-- Also, please read the contribution guidelines: https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md -->

thomas-lebeau commented on packages/flagging/package.json:
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "@datadog/browser-flagging",
+  "version": "6.6.4",
```
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏ This should be in sync with the the latest version (`6.7.0`). Our release process will keep it in sync automatically at every release

thomas-lebeau commented on packages/flagging/package.json:
```diff
@@ -0,0 +1,38 @@
+    "replace-build-env": "node ../../scripts/build/replace-build-env.js"
+  },
+  "dependencies": {
+    "@datadog/browser-core": "6.6.4"
```
```suggestion
    "@datadog/browser-core": "6.7.0"
```

thomas-lebeau commented on packages/flagging/package.json:
```diff
@@ -0,0 +1,38 @@
+    "@datadog/browser-core": "6.6.4"
+  },
+  "peerDependencies": {
+    "@datadog/browser-rum": "6.6.4"
```
```suggestion
    "@datadog/browser-rum": "6.7.0"
```

thomas-lebeau reviewed (commented):
Could you also add `flagging` to the list of packages [here](https://github.com/DataDog/browser-sdk/blob/61824b8d170a092c20b3a1006841031b935c30e5/scripts/lib/computeBundleSize.js#L6), this will add an entry to the bundle size PR comment (the CI job is a bit flaky so it does not always happens, but we'll work on that)

Finally, following [our convention](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md#commit-messages-and-pull-request-titles), the PR title would be "👷 [FFL-23] Create datadog-flagging package". This convention is used to generate the changelog

leoromanovsky commented:
> Finally, following [our convention](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md#commit-messages-and-pull-request-titles), the PR title would be "👷 [FFL-23] Create datadog-flagging package". This convention is used to generate the changelog

Understood - in the future I'll make the first commit in a branch follow that format so that the PR title and squashed commit comply with it.

I pushed a throwaway commit with the same title. Generally I don't like reordering commits in PRs once reviews have started to preserve the timeline. In this case would you like me to squash into one commit or rename the first commit so that `👷 [FFL-23] Create datadog-flagging package` shows up in the git log?

aarsilv commented on .github/CODEOWNERS:
```diff
@@ -1,7 +1,8 @@
 
 # Global
-*             @Datadog/rum-browser
+packages/flagging  @Datadog/feature-flagging-experimentation
```
🎉

aarsilv commented on .github/codeql-config.yml:
```diff
@@ -3,6 +3,7 @@ name: 'CodeQL config'
 paths:
   - packages/core/src
   - packages/logs/src
+  - packages/flagging/src
```
📈

aarsilv commented on packages/flagging/package.json:
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "@datadog/browser-flagging",
+  "version": "6.7.0",
```
This is because at Datadog the convention is for all versions that play together to be kept in sync?

BenoitZugmeyer replied:
It's not a Datadog convention, more like our team convention 😄 

The version will be automatically updated on every release.

aarsilv commented on packages/flagging/src/hello.ts:
```diff
@@ -0,0 +1,7 @@
+export const flagging = {
+  init: () => {
+    // TODO: Implement initialization logic
```
😄

aarsilv commented on sandbox/react-app/main.tsx:
```diff
@@ -4,13 +4,16 @@ import ReactDOM from 'react-dom/client'
   plugins: [reactPlugin({ router: true })],
 })
 
+datadogFlagging.init()
```
💪

aarsilv reviewed (approved):
📈 📈 📈

BenoitZugmeyer commented on packages/flagging/README.md:
```diff
@@ -0,0 +1,3 @@
+# Flagging SDK
+
+This package supports flagging and experimentation by performing evaluation in the browser.
```
**suggestion:** mention that this package is not yet intended for public usage?

BenoitZugmeyer commented on packages/flagging/package.json:
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "@datadog/browser-flagging",
+  "version": "6.7.0",
```
**suggestion:** set `"private": true` to avoid publishing it to NPM yet?

leoromanovsky commented:
👋 Hey folks I have addressed all feedback in the PR and am ready to merge, but github is blocking because `Commits must have verified signatures.`. I have now completed the process and pushed an empty signed commit but this does not unblock merging.

Normally I never change the git timeline in a PR but in this case I need to either resign all commits locally or squash them into one signed commit; either way there will be a force push to this branch.

leoromanovsky commented:
Yes - when it’s time for public release we can engage the product or
marketing team for naming input on a memorable name, I just needed a
reasonable container of code to develop against. Do you have an input now?

On Fri, May 23, 2025 at 10:28 AM Eric Petzel ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In eslint.config.mjs
> <https://github.com/DataDog/browser-sdk/pull/3553#discussion_r2105105302>:
>
> > @@ -341,7 +341,7 @@ export default tseslint.config(
>    },
>
>    {
> -    files: ['packages/{rum,logs,rum-slim}/src/entries/*.ts'],
>
> Is the intent to put experiment-specific code (like exposure
> callbacks/logging) in this package as well? If so, is flagging our
> preferred option for naming?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/DataDog/browser-sdk/pull/3553#pullrequestreview-2865191817>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAOAELU5CIITL2BW6YG64L275LFHAVCNFSM6AAAAAB5U7UVMSVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDQNRVGE4TCOBRG4>
> .
> You are receiving this because you were assigned.Message ID:
> ***@***.***>
>

leoromanovsky commented on eslint.config.mjs:
```diff
@@ -341,7 +341,7 @@ export default tseslint.config(
   },
 
   {
-    files: ['packages/{rum,logs,rum-slim}/src/entries/*.ts'],
```
Yes - when it’s time for public release we can engage the product or marketing team for naming input on a memorable name, I just needed a reasonable container of code to develop against. Do you have an input now?
